x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel_wrapper.test.tsx

```diff
@@ -7,67 +7,106 @@
 
 import React from 'react';
 import { Visualization } from '../../../types';
-import { createMockVisualization, createMockFramePublicAPI, FrameMock } from '../../../mocks';
-import { WorkspacePanelWrapper } from './workspace_panel_wrapper';
-import { mountWithProvider } from '../../../mocks';
-import { ReactWrapper } from 'enzyme';
 import {
-  selectAutoApplyEnabled,
-  updateVisualizationState,
-  disableAutoApply,
-  selectTriggerApplyChanges,
-} from '../../../state_management';
-import { enableAutoApply, setChangesApplied } from '../../../state_management/lens_slice';
+  createMockVisualization,
+  createMockFramePublicAPI,
+  FrameMock,
+  renderWithReduxStore,
+} from '../../../mocks';
+import { WorkspacePanelWrapper } from './workspace_panel_wrapper';
+import { updateVisualizationState, LensAppState } from '../../../state_management';
+import { setChangesApplied } from '../../../state_management/lens_slice';
 import { LensInspector } from '../../../lens_inspector_service';
+import { screen } from '@testing-library/react';
+import faker from 'faker';
+import { SettingsMenu } from '../../../app_plugin/settings_menu';
 
 describe('workspace_panel_wrapper', () => {
   let mockVisualization: jest.Mocked<Visualization>;
   let mockFrameAPI: FrameMock;
+  const ToolbarComponentMock = jest.fn(() => null);
+
+  const renderWorkspacePanelWrapper = (
+    propsOverrides = {},
+    { preloadedState }: { preloadedState: Partial<LensAppState> } = { preloadedState: {} }
+  ) => {
+    const { store, ...rtlRender } = renderWithReduxStore(
+      <>
+        <WorkspacePanelWrapper
+          framePublicAPI={mockFrameAPI}
+          visualizationState={{ internalState: 123 }}
+          visualizationId="myVis"
+          visualizationMap={{
+            myVis: { ...mockVisualization, ToolbarComponent: ToolbarComponentMock },
+          }}
+          datasourceMap={{}}
+          datasourceStates={{}}
+          isFullscreen={false}
+          lensInspector={{} as unknown as LensInspector}
+          getUserMessages={() => []}
+          children={<span />}
+          {...propsOverrides}
+        />
+        <SettingsMenu
```

Comment:
instead of interacting with the store, I've mounted the component that toggles autoapply button. WDYT about this approach?

Reply:
Cool idea. I like that this turns it into more of an integration test and is closer to how users experience the app (UI as opposed to programmatic invocations).